src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java

```diff
@@ -39,17 +40,14 @@ public void can_invoke_late_assertion_on_assertThatCode() {
   @Test
   public void should_fail_when_asserting_no_exception_raised_but_exception_occurs() {
     // Given
-    ThrowingCallable boom = raisingException("boom");
+    Exception exception = new Exception("boom");
+    ThrowingCallable boom = raisingException(exception);
 
     // Expect
-    thrown.expectAssertionError(format("[Test] %n" +
-                                       "Expecting code not to raise a throwable but caught a%n" +
-                                       "  <java.lang.Exception>%n" +
-                                       "with message :%n" +
-                                       "  \"boom\""));
+    thrown.expectAssertionError(ShouldNotHaveThrown.shouldNotHaveThrown(exception));
```

Comment:
static import `shouldNotHaveThrown`

Reply:
add a separate test to verify the error message content as in https://github.com/joel-costigliola/assertj-core/blob/08c276844c55e74b5773248e7c5c9b6a741a4b4d/src/test/java/org/assertj/core/error/ShouldBeBefore_create_Test.java

Reply:
I wasn't sure of the best way to test both the description with `as()` and also testing the "stacktrace" way without making the test brittle - I'll take another look at the stacktrace case and maybe I can just assert that the message begins with the description in the `as()` test case? WDYT?

Reply:
`startsWith` and `contains` String assertions should do the job.